### ISSUE TITLE
es-ES audio forced like ntflx addon

### ISSUE THREAD

**Issue by DanilosTeam** [Sandmann79/xbmc#262]
With Spanish select on language player settings   the movies are played by default in Latin American spanish.
I think the Spanish of Spain es-ES can be forced  by modifying the addon code.
In Netflix addon the developer fix It here:
https://github.com/asciidisco/plugin.video.netflix/issues/580
Can we do this on Amazon vod addon?

**Comment by Essam315** [comment]
This issue explains the problem of Kodi and the European Spanish (Spanish of Spain).

https://github.com/xbmc/xbmc/issues/15308

**Comment by Varstahl** [comment]
A thing that I noticed fast is that in regard to l10n and i18n, is that Kodi sucks quite a bit. Replacing Latin American spanish with european spanish is not necessarily the best path to go down, just as prioritizing portuguese over brazilian portuguese, or one of the nederlaandese languages over the other. This could (and should) be fixed at a Kodi level, but I can see why this is an issue for a lot of people.

It's possible, we already do some slight locale modifications, but just let me think this through before patching things randomly. I could have something working early next week.

**Comment by DanilosTeam** [comment]
Thanks for the reply. I hope it can be solved. This addon is very useful for many users!

**Comment by Varstahl** [comment]
For a moment there I miscalculated, and I believed we were talking subtitles. After checking that out I noticed that, to the contrary, we were talking audio streams embedded inside .MPDs we pass straight to IS.A. The Netflix addon does things a bit differently, and that's why they can.

It is technically possible to do so, but we would need a sort of "call proxy" to do that. I was working on a cryptoproxy for the addon so I might reuse that, but this will take more time and testing than anticipated. Don't expect this shortly.

**Comment by DanilosTeam** [comment]
Okay. But it is interesting to know how uncomfortable it is to change the audio language to all the chapters of a TV movie, for example Jack Ryan, from Latin to European Spanish. I think that non-Spaniards can not imagine how unpleasant is the Latin audio for the user from Spain. Thank you very much for your work and congratulations for the addon.

**Comment by Varstahl** [comment]
That's where amazon fucked up a little:
```
<AdaptationSet audioTrackId="es-es_dialog_0" contentType="audio" group="1" lang="es" <AdaptationSet audioTrackId="es-419_dialog_0" contentType="audio" group="1" lang="es"
```
I'm sure they have a good reason to drop the locale in the language field, but the problem for Kodi is that now everything looks like spanish to it. I'm not even sure if this can be fixed. It should be possible through a proxy and some manipulation, but I'm uncertain if changing the languages from es to es_ES and es_LA would make any difference for Kodi. We'll see.

**Comment by Essam315** [comment]
@Varstahl
You could try to use this workaround change the Latin American if possible from 'es' to 'eo', the code of 'Esperanto' and no one is using this language, but this will be a fix for the OP only.

**Comment by Varstahl** [comment]
Once we can manage to change the language code we could even suppress them, that's not the problem. I'm currently working on the proxy, recycling some of the stuff I previously did for a cryptoproxy, I'll let you know soon.

**Comment by Varstahl** [comment]
You can find [here](https://github.com/Varstahl/xbmc/tree/add-proxy-service) the first implementation of the proxy server for MPD alteration. As per already established subtitle rules, `es-es` is converted into `es`, and `es-419` into `es-Latinoamerica`. This is just a first implementation, and we can do more, but in my tests it should already work for @DanilosTeam. Could you download the `plugin.video.amazon-test` from the repository I linked and give it a test?

**Comment by DanilosTeam** [comment]
Thank you @Varstahl !!!!
When I get home tonight I test it and post it.

**Comment by Essam315** [comment]
@Varstahl
I tested the audio with "Jack Ryan", "Seinfeld" and  "Community", and the "European Spanish" is selected by default, when i choose "Spanish" in "Preferred audio language".

**Comment by DanilosTeam** [comment]
The change is the addition of proxy.py?

Thx!!! @Varstahl & @Essam315

**Comment by Essam315** [comment]
@DanilosTeam
Here is the new file and the changes in the old files.

https://github.com/Varstahl/xbmc/commit/6b99251d2a3fca8674f770a1b41b359df3819dda

**Comment by Varstahl** [comment]
> The change is the addition of proxy.py?

@DanilosTeam, as @Essam315 pointed out, you need to create `proxy.py` and replace `service.py`, `common.py` and `playback.py`. Or, just download the zipped version of the repository, and unpack the amazon-test directory. Whatever works for you.

**Comment by DanilosTeam** [comment]
New implementation works perfect on my system. Tested with Jack Ryan episode. Before uninstall previous add-on, I've got "Latinoamericano" by default. When I install modify add-on the same episode has got es-ES by default. 

Thanks everybody for this great work. Thanks for all!

**Comment by DanilosTeam** [comment]
@Essam315 I'll have to make libreelec fresh install on a Pi. Then I install script.module.pydes 
Thanks!

**Comment by Paco8** [comment]
There's a small problem. If a video has only a latin american Spanish audio track (and not European Spanish), then the addon doesn't select the latin american Spanish by default as one would expect, it seems it selects the first audio track, which may be German. This can be tested with the show Parks and Recreation, it has several audio languages, but the Spanish one is the latin american.

**Comment by Varstahl** [comment]
The addon doesn't choose anything, Kodi does. There might be reasons why a language is selected over the other. Keep in mind though, that I'm still working on the proxy affaire, it's going to take a while to iron things out.

**Comment by Varstahl** [comment]
I just checked and you're absolutely correct. And yes, it does. The problem is that I can't really do much about it. I thought about reordering, thinking that maybe I could impose a strict order of preference just as I did for subtitles, but that doesn't work, and german starting with a d (Deutsch) gets always first and selected. Same goes for subtitles. The only thing I could do is to try and suppress the locale if it's the only language available. It's far from perfect but there's not much else I can do (for the moment).

I don't even know if we should bring this up with the Kodi team. I now understand though why some peeps from the UK were bothered by the fact that stuff always started in german. They're both coded en_GB and en_US, so neither of those would start in subtitles and the like.

What a fine mess…

**Comment by Paco8** [comment]
I guess Kodi should at least allow to select between European Spanish and Latin Spanish as preferred language.

Now I just noticed that selecting English as preferred audio language doesn't work. It plays in German or Arabic. I guess this because the English track is labeled as "English - US" instead of just "English".

I also found in a movie the language "French - Catalan". I guess it's actually "French - Canada".
